hack/bundle-gen.py

```diff
@@ -129,26 +130,59 @@ def get_params():
 
     return args
 
+# Traverse through version directories to detect highest version
+def get_previous_version(work_dir, channel_name):
+    upstream_repo = "git@github.com:redhat-openshift-ecosystem/community-operators-prod.git"
+    upstream_branch = "main"
+    dir_name = "prev_version_determiner"
 
-# get_previous_version grabs the previous hive version (without the leading `v`) from
-# COMMUNITY_OPERATORS_HIVE_PKG_URL package yaml for the provided channel_name.
-def get_previous_version(channel_name):
-    http = urllib3.PoolManager()
-    r = http.request("GET", COMMUNITY_OPERATORS_HIVE_PKG_URL)
-    pkg_yaml = yaml.load(r.data.decode("utf-8"), Loader=yaml.FullLoader)
+    repo_full_path = os.path.join(work_dir, dir_name)
+    # clone git repo
+    try:
+        git.Repo.clone_from(upstream_repo, repo_full_path)
```

Comment:
We're already cloning this repo in open_pr. Let's find a way to collapse and only do it once.